### PR TITLE
Rebase to 0.4

### PR DESCRIPTION
**Pull request by bilelmoussaoui**
cc @sdroege @jf2048 

Will do a new release once this lands

### CODE REVIEW

**Comment by bilelmoussaoui** [comment]
I wonder if the test attribute macro requires a recent rustc

**Comment by jf2048** [comment]
That's an odd error, let me do some digging

File: gtk4-macros/src/lib.rs [hunk 1]
```diff
@@ -256,3 +272,67 @@ pub fn template_callbacks(attr: TokenStream, item: TokenStream) -> TokenStream {
...
+            quote! {
+                #(#attrs)*
+                #[::std::prelude::v1::test]
+                #vis #sig {
+                    #crate_ident::test_synced(move || {
+                        #block
+                    })
+                }
+            }
+            .into()
```
Comment (jf2048) [inline gtk4-macros/src/lib.rs:334]:
```suggestion
            let test = quote! {
                #(#attrs)*
                #[::std::prelude::v1::test]
                #vis #sig {
                    #crate_ident::test_synced(move || {
                        #block
                    })
                }
            };
            test.into()
```

I'm not sure why, but this seems to fix it

**Review by jf2048 — CHANGES_REQUESTED** [review]

File: gtk4-macros/src/lib.rs (hunk 1 above)
Comment (jf2048) [reply gtk4-macros/src/lib.rs:334]: #977

Comment (bilelmoussaoui) [reply gtk4-macros/src/lib.rs:334]: Awesome, thanks

Comment (bilelmoussaoui) [reply gtk4-macros/src/lib.rs]: >  It doesn't affect that because we've bumped the version to rustc 1.57.0 

There is no need for that I would say

**Pull request merged**